### PR TITLE
Update RebuildBenchmarkHistory.yaml

### PR DESCRIPTION
**Pull request by vmelamed**
# Summary

update RebuildBenchmarkHistory.yaml (it was only a temporary one)

## Commits

- feat(workflow): update RebuildBenchmarkHistory.yaml with improved comments and remove unused matrix strategy

## Checklist

- [x] Commit messages follow [Conventional Commits](https://www.conventionalcommits.org/) format
- [x] Tests pass locally, and cover new code and edge cases. Test coverage is above 80%.
- [ ] It is not expected that the performance will degrade by more than 20%
- [x] Documentation is updated if necessary

### CODE REVIEW

File: .github/workflows/RebuildBenchHistory.yaml [hunk 1]
```diff
@@ -1,5 +1,10 @@
 name: "Rebuild Benchmark History"
 
+# Manually triggered (UI/phone) per-repo entry point. Re-records this repo's benchmark history to Bencher.dev by running
+# every benchmark project under benchmarks/ N times (record-only: no thresholds, no failure on a noisy point). Use after a
+# runner-image change or a benchmark restructure invalidates the existing baseline. Shared, identical across repos -- a repo
+# with no benchmarks/ simply has nothing to do.
```
[the remark this answers is not in the document]
Comment (vmelamed) [reply .github/workflows/RebuildBenchHistory.yaml]: edited
